src/wasm/wasm-stack.cpp

```diff
@@ -2033,15 +2033,14 @@ void BinaryInstWriter::visitRefCast(RefCast* curr) {
   o << int8_t(BinaryConsts::GCPrefix);
   if (curr->safety == RefCast::Unsafe) {
     o << U32LEB(BinaryConsts::RefCastNop);
-    parent.writeIndexedHeapType(curr->intendedType);
+    parent.writeIndexedHeapType(curr->type.getHeapType());
```

Comment:
Do we not need to check if the type is a reference here? Unreachable would be a problem. (if so, then below as well on 2043).

Reply:
No, since we never emit instructions with unreachable children.